lightblue-migration-monitor/src/main/java/com/redhat/lightblue/migrator/monitor/Monitor.java

```diff
@@ -0,0 +1,93 @@
+import com.redhat.lightblue.client.LightblueException;
+import com.redhat.lightblue.client.Literal;
+import com.redhat.lightblue.client.Projection;
+import com.redhat.lightblue.client.Query;
+import com.redhat.lightblue.client.http.LightblueHttpClient;
+import com.redhat.lightblue.client.request.data.DataFindRequest;
+import com.redhat.lightblue.client.response.LightblueDataResponse;
+import com.redhat.lightblue.migrator.MigrationConfiguration;
+import com.redhat.lightblue.migrator.MigrationJob;
+
+public class Monitor {
+
+    private final LightblueClient lightblueClient;
+
+    public Monitor(MonitorConfiguration cfg) {
+        if (cfg.getClientConfig() != null) {
+            lightblueClient = new LightblueHttpClient(cfg.getClientConfig());
+        } else {
+            lightblueClient = new LightblueHttpClient();
+        }
+    }
+
+    public void runCheck(final Notifier... notifiers) throws LightblueException {
+        List<String> configurationsMissingJobs = new ArrayList<>();
+        for (MigrationConfiguration cfg : findMigrationConfigurations()) {
+            long period = parsePeriod(cfg.getPeriod());
+            Date endDate = new Date();
+            Date startDate = new Date(endDate.getTime() - period);
+
+            int jobs = countMigrationJobs(cfg.getConfigurationName(), startDate, endDate);
+            if (jobs <= 0) {
+                configurationsMissingJobs.add(cfg.getConfigurationName());
+            }
+        }
+
+        if (configurationsMissingJobs.isEmpty()) {
+            for (Notifier n : notifiers) {
+                n.sendSuccess();
+            }
+        } else {
+            for (Notifier n : notifiers) {
+                n.sendFailure(configurationsMissingJobs);
+            }
+        }
+    }
+
+    /**
+     * Returns the period in msecs
+     */
+    private static long parsePeriod(String periodStr) {
+        PeriodFormatter fmt = PeriodFormat.getDefault();
+        Period p = fmt.parsePeriod(periodStr);
+        return p.toStandardDuration().getMillis();
+    }
+
+    private MigrationConfiguration[] findMigrationConfigurations() throws LightblueException {
+        DataFindRequest findConfigurations = new DataFindRequest(MigrationConfiguration.ENTITY_NAME);
+        findConfigurations.where(Query.withValue("period", Query.neq,
+                Literal.pojo(null)));
+        findConfigurations.select(
+                Projection.includeField("configurationName"),
+                Projection.includeField("period"));
+
+        return lightblueClient.data(findConfigurations, MigrationConfiguration[].class);
+    }
+
+    private int countMigrationJobs(String configurationName, Date startDate, Date endDate) throws LightblueException {
+        DataFindRequest findJobs = new DataFindRequest(MigrationJob.ENTITY_NAME);
+        findJobs.where(
+            Query.and(
+                Query.withValue("configurationName", Query.eq, configurationName),
```

Comment:
Your concerns are valid. This can only be used if the ranges use a know
date syntax, and that those ranges are really dates. But what is the
alternative? Those fields are the only ones that tell what range the job is
supposed to process.

On Tue, May 24, 2016 at 6:59 AM, Dennis Crissman <notifications@github.com>
wrote:

> In
> lightblue-migration-monitor/src/main/java/com/redhat/lightblue/migrator/monitor/Monitor.java
> <https://github.com/lightblue-platform/lightblue-migrator/pull/356#discussion_r64385350>
> :
>
> > +    private MigrationConfiguration[] findMigrationConfigurations() throws LightblueException {
> > +        DataFindRequest findConfigurations = new DataFindRequest(MigrationConfiguration.ENTITY_NAME);
> > +        findConfigurations.where(Query.withValue("period", Query.neq,
> > +                Literal.pojo(null)));
> > +        findConfigurations.select(
> > +                Projection.includeField("configurationName"),
> > +                Projection.includeField("period"));
> > +
> > +        return lightblueClient.data(findConfigurations, MigrationConfiguration[].class);
> > +    }
> > +
> > +    private int countMigrationJobs(String configurationName, Date startDate, Date endDate) throws LightblueException {
> > +        DataFindRequest findJobs = new DataFindRequest(MigrationJob.ENTITY_NAME);
> > +        findJobs.where(
> > +            Query.and(
> > +                Query.withValue("configurationName", Query.eq, configurationName),
>
> I hear what you are saying about wanting to use the the jobRange fields,
> but I still have some concerns about it.
>
> Firstly, that would then be doing a string compare instead of a date
> compare, which could produce different results.
> Secondly, there is no way to guarantee the date format used in the string
> field.
> Thirdly, there is no way to know for sure that the string field even
> contains a date and not some other sort of value. You said yourself it
> could be an id field.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-migrator/pull/356/files/d9ae8a0dbee0850dc48d9d901a915525c4540c70#r64385350>
>